### PR TITLE
Fix tests on windows

### PR DESCRIPTION
**Pull request by theweavrs**
Summary:
---------
This fixes all the failing tests on Windows.

Test Plan:
----------
Ran `yarn test` on Windows and everything passed.

### CODE REVIEW

**Comment by thymikee** [comment]
Looks like we need to sort the output in the init e2e test

**Comment by theweavrs** [comment]
The tests that check `templateFiles` fail due to incorrect order of files in the directory. Other fail due to incorrect output.

**Comment by thymikee** [comment]
Last failure left

File: packages/cli/src/tools/__tests__/copyFiles.test.js [hunk 1]
```diff
@@ -47,10 +51,10 @@ test('copies text and binary files from source to destination', async () => {
 
 test('copies files from source to destination excluding directory', async () => {
   const src = path.resolve(__dirname, './__fixtures__');
+  let regexStr = path.join(src, 'extraDir');
   await copyFiles(src, DIR, {
-    exclude: [new RegExp(path.join(src, 'extraDir'))],
+    exclude: [new RegExp(RegexEscape(regexStr))],
```
Comment (thymikee) [inline packages/cli/src/tools/__tests__/copyFiles.test.js]: We use it as `exclude: [new RegExp(path.resolve(templatePath, 'node_modules'))]` in the source. Maybe it's not working on Windows? Could you verify on a local template and see if `node_modules` are copied? They shouldn't

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: Yes they are being copied. Tried on local and remote template.

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: I can't `init` using a template. Remote or local.

**For local I get this error:**
```
error Error: Cannot find module '\C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\react-native-template-typescript-master-cli-v2\package.json'
error EBUSY: resource busy or locked, rmdir 'C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\TestInit2'
Error: EBUSY: resource busy or locked, rmdir 'C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\TestInit2'
    at Object.rmdirSync (fs.js:684:3)
    at rmdirSync (C:\Users\thecodrr\Source\Repos\cli\node_modules\fs-extra\lib\remove\rimraf.js:276:13)
    at Object.rimrafSync [as removeSync] (C:\Users\thecodrr\Source\Repos\cli\node_modules\fs-extra\lib\remove\rimraf.js:252:7)
    at Object.initialize [as func] (C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\commands\init\init.js:295:26)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:832:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

For Remote:
```
debug Installing template from typescript
yarn add v1.16.0
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ typescript@3.5.3
info All dependencies
└─ typescript@3.5.3
Done in 7.86s.
debug Getting config from C:\Users\thecodrr\AppData\Local\Temp\rncli-init-template-jV11JR\node_modules\typescript\template.config.js
error Error: Cannot find module 'C:\Users\thecodrr\AppData\Local\Temp\rncli-init-template-jV11JR\node_modules\typescript\template.config'
error EBUSY: resource busy or locked, rmdir 'C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\TestInit2'
Error: EBUSY: resource busy or locked, rmdir 'C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\TestInit2'
    at Object.rmdirSync (fs.js:684:3)
    at rmdirSync (C:\Users\thecodrr\Source\Repos\cli\node_modules\fs-extra\lib\remove\rimraf.js:276:13)
    at Object.rimrafSync [as removeSync] (C:\Users\thecodrr\Source\Repos\cli\node_modules\fs-extra\lib\remove\rimraf.js:252:7)
    at Object.initialize [as func] (C:\Users\thecodrr\Source\Repos\cli\packages\cli\build\commands\init\init.js:295:26)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
In remote, the temp directory is deleted before it can copy the template (don't know why). And in local, the path is always wrong.

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: No they are not being deleted

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: Tests should be fixed now.

Comment (thymikee) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: I'm not sure if I follow. The point of this is to exclude certain dirs, like `node_modules`. If you have a local template with `node_modules` inside, they _shouldn't_ be copied to the new project. From the code you added, it seems that this escaping needs to be done in the source, not in the test only, because this is exactly how we use it. Can you move the escape fn there?

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: I did create an escape function in the source; check the latest commits.

Sure, I will rename it.

Comment (theweavrs) [reply packages/cli/src/tools/__tests__/copyFiles.test.js]: There is no need for copying as `jest-regex-util` is already present, we can just use the function from there.

File: packages/cli/src/commands/init/template.js [hunk 2]
```diff
@@ -4,10 +4,7 @@ import path from 'path';
 import * as PackageManager from '../../tools/packageManager';
 import {logger} from '@react-native-community/cli-tools';
 import copyFiles from '../../tools/copyFiles';
-
-const RegexEscape = function(regex) {
-  return regex.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
-};
+import {replacePathSepForRegex} from 'jest-regex-util';
```
Comment (thymikee) [inline packages/cli/src/commands/init/template.js]: It's in our development deps. Jest libs will usually be there but there's not guarantee. If we import a package, we need to declare it as our dependency. So, either copy the oneliner from jest-regex-util or add it to the deps of packages/cli/package.json. I'd prefer to copy to avoid users having multiple versions of jest packages :)

Comment (theweavrs) [reply packages/cli/src/commands/init/template.js]: Oh okay, got it. I will make a copy.

File: packages/cli/src/commands/init/init.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ async function setProjectDirectory(directory) {
       throw new DirectoryAlreadyExistsError(directory);
     }
 
-    await fs.emptyDir(directory);
+    fs.emptyDirSync(directory);
```
Comment (thymikee) [inline packages/cli/src/commands/init/init.js]: I wonder, why this change?

Comment (theweavrs) [reply packages/cli/src/commands/init/init.js]: Using await causes hang on Windows, that's why.

Comment (thymikee) [reply packages/cli/src/commands/init/init.js]: Geez, Windows is so weird

Comment (theweavrs) [reply packages/cli/src/commands/init/init.js]: It indeed is. 😂😂
